modules/core/src/main/java/org/apache/ignite/internal/processors/datastreamer/DataStreamerImpl.java

```diff
@@ -837,20 +845,13 @@ private void doFlush() throws IgniteCheckedException {
 
         List<IgniteInternalFuture> activeFuts0 = null;
 
-        int doneCnt = 0;
-
         for (IgniteInternalFuture<?> f : activeFuts) {
             if (!f.isDone()) {
                 if (activeFuts0 == null)
-                    activeFuts0 = new ArrayList<>((int)(activeFuts.size() * 1.2));
+                    activeFuts0 = new ArrayList<>(activeFuts.size() / 2);
 
                 activeFuts0.add(f);
             }
-            else {
-                f.get();
```

Comment:
You mustn't delete this call (f.get()) because otherwise the user code, that called flush() or close() methods, can miss an Exception that happened during a pre-loading of a batch of data.

Actually current logic is supposed to store the flushing as soon as at least one exception occurs. We can leave it as is for now.